hazelcast/src/main/java/com/hazelcast/query/impl/Index.java

```diff
@@ -17,16 +17,23 @@
 package com.hazelcast.query.impl;
 
 import com.hazelcast.core.TypeConverter;
-import com.hazelcast.nio.serialization.Data;
+import com.hazelcast.map.impl.record.Record;
 import com.hazelcast.query.QueryException;
 
 import java.util.Set;
+
 /**
  * This interface contains the methods related to index of Query.
  */
 public interface Index {
 
-    void saveEntryIndex(QueryableEntry e) throws QueryException;
+    /**
+     * Add entry to this index.
```

Comment:
What happens when an item for the given key already exist?

And can you add some documentation why the old value needs to be passed? I'm sure it is needed, but when I look at the API it doesn't make much sense.